src/azul/indexer/document.py

```diff
@@ -268,21 +286,62 @@ def to_index(self, value: T) -> T:
     def from_index(self, value: T) -> T:
         return value
 
+    @property
+    def operators(self) -> Tuple[str, ...]:
+        if self.native_type == int:
+            return 'is', 'within'
+        else:
+            return 'is',
 
-pass_thru_str: PassThrough[str] = PassThrough(es_type='keyword')
-pass_thru_int: PassThrough[int] = PassThrough(es_type='long')
-pass_thru_bool: PassThrough[bool] = PassThrough(es_type='boolean')
+
+pass_thru_str: PassThrough[str] = PassThrough(str, es_type='keyword')
+pass_thru_int: PassThrough[int] = PassThrough(int, es_type='long')
+pass_thru_bool: PassThrough[bool] = PassThrough(bool, es_type='boolean')
 # FIXME: change the es_type for JSON to `nested`
 #        https://github.com/DataBiosphere/azul/issues/2621
-pass_thru_json: PassThrough[JSON] = PassThrough(es_type=None)
+pass_thru_json: PassThrough[JSON] = PassThrough(JSON, es_type=None)
+
+
+class Range(PassThrough[JSON]):
+    operators = ('is', 'within', 'contains', 'intersects')
+
+
+range_ = Range(JSON, es_type=None)
```

Comment:
```suggestion
range_ = ClosedRange(JSON, es_type=None)
```

Reply:
And wouldn't we want to pin `es_type` at the class definition as opposed to at every instantiation?

Reply:
from PL: moved es_type to `__init__`

Reply:
from PL: moved es_type to `__init__`